script.js

```diff
@@ -0,0 +1,155 @@
+const apiKey = 'DBELFYET2HB66AN6SZC56BC6M';
+const form = document.getElementById('search-form');
+const input = document.getElementById('location-input');
+const weatherDisplay = document.getElementById('weather-display');
+const forecastDisplay = document.getElementById('forecast-display');
+const toggleButton = document.getElementById('toggle-units');
+const loadingIndicator = document.getElementById('loading');
+const errorMessage = document.getElementById('error-message');
+
+let currentUnit = 'metric'; // metric = Celsius, us = Fahrenheit
+let lastLocation = '';
+
+form.addEventListener('submit', (e) => {
+    e.preventDefault();
+    const location = input.value.trim();
+    if (location) {
+        clearDisplays();
+        showLoading();
+        fetchWeatherData(location, currentUnit)
+        .then(data => {
+            hideLoading();
+            processAndDisplayWeather(data, currentUnit);
+            lastLocation = location;
+        })
+        .catch(err => {
+            hideLoading();
+            showError('Location not found. Please try again.');
+        });
+    }
+})
+
+toggleButton.addEventListener('click', () => {
+    currentUnit = currentUnit === 'metric' ? 'us' : 'metric';
+    toggleButton.textContent = currentUnit === 'metric' ? 'Switch to °F' : 'Switch to °C';
+    if (lastLocation) {
+        clearDisplays();
+        showLoading();
+        fetchWeatherData(lastLocation, currentUnit)
+        .then(data => {
+            hideLoading();
+            processAndDisplayWeather(data, currentUnit);
+        })
+        .catch(err => {
+            hideLoading();
+            showError('Location not found. Please try again.');
+        });
+    }
+});
+
+//      API FETCH FUNCTION
+async function fetchWeatherData(location, unitGroup) {
+    const url = `https://weather.visualcrossing.com/VisualCrossingWebServices/rest/services/timeline/${encodeURIComponent(location)}?unitGroup=${unitGroup}&key=${apiKey}&include=current,days&elements=datetime,temp,humidity,windspeed,precipprob,conditions,icon`;
+    const response = await fetch(url);
```

Comment:
You can also add loading time out to prevent the app from hanging indefinitely on slow network.